build/docker/scripts/ubsan.sh

```diff
@@ -15,7 +15,7 @@ export CXX=clang++-3.8
 # undefined casting, aka "vptr".
 #
 # TODO: fix undefined vptr behavior and turn this option back on.
-export CFLAGS="-fsanitize=undefined -fno-sanitize-recover=undefined -fno-sanitize=vptr"
```

Comment:
I should have been more explicit. For finding `vptr` errors, it's good to remove `-fno-sanitize-recover=undefined`. But for checking that there are none, it's good to have `-fno-sanitize-recover=undefined`.

If this patch fixes some, but not all, of the `vptr` errors, I would suggest leaving `CFLAGS` as it is in `master`. If it fixes all `vptr` errors, I think the way you have it now is optimal.